src/mlpack/methods/ne/cne.hpp

```diff
@@ -0,0 +1,186 @@
+ /**
+ * @file cne.hpp
+ * @author Bang Liu
+ *
+ * Definition of CNE class.
+ */
+#ifndef MLPACK_METHODS_NE_CNE_HPP
+#define MLPACK_METHODS_NE_CNE_HPP
+
+#include <cstddef>
+#include <cstdio>
+
+#include <mlpack/core.hpp>
+
+#include "link_gene.hpp"
+#include "neuron_gene.hpp"
+#include "genome.hpp"
+#include "species.hpp"
+#include "tasks.hpp"
+#include "parameters.hpp"
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This class implements Conventional Neuro-evolution (CNE): weight
+ * evolution on topologically fixed neural networks.
+ */
+template<typename TaskType>
+class CNE {
+ public:
+  // Parametric constructor.
+  CNE(TaskType task, Genome& seedGenome, Parameters& params) {
```

Comment:
I think for algorithms the parameters are a lot. Users can set parameter by construct an Parameters object without put them into a constructor function in a specific sequence. And this also makes the algorithm constructor be concise. If we have more parameters in the future, we just add new member for Parameters class, no need to change the function headers of constructors. Besides, different NE algorithm will have similar functions for constructor.